src/frontend/tests/e2e/webDriverPageObjectElement.ts

```diff
@@ -0,0 +1,113 @@
+import { PageObjectElement } from "../../types/pageObject";
+import { isNullish } from "@dfinity/utils";
+
+export class WebDriverPageObjectElement implements PageObjectElement {
+  readonly #browser: WebdriverIO.Browser;
+  readonly #selectElement: () => Promise<WebdriverIO.Element | undefined>;
+  #element?: Promise<WebdriverIO.Element>;
+
+  constructor(
+    browser: WebdriverIO.Browser,
+    selectElement: () => Promise<WebdriverIO.Element | undefined>,
+  ) {
+    this.#browser = browser;
+    this.#selectElement = selectElement;
+  }
+
+  static create(browser: WebdriverIO.Browser): PageObjectElement {
+    return new WebDriverPageObjectElement(browser, () => browser.$("body"));
+  }
+
+  #getElement(): Promise<WebdriverIO.Element> {
+    if (isNullish(this.#element)) {
+      this.#element = this.#browser
+        .waitUntil(this.#selectElement, {
+          timeout: 10000,
+        })
+        .then((element) => element!);
+    }
+    return this.#element;
+  }
+
+  getByRole(role: string, { name }: { name: string }): PageObjectElement {
+    return new WebDriverPageObjectElement(this.#browser, async () => {
+      // WebdriverIO doesn't support combining both text and role,
+      // so instead we select all text matches and then check for
+      // the matching role with the `getComputedRole` method.
+      const elements = await (await this.#getElement()).$$(`aria/${name}`);
+      for (const element of elements) {
+        try {
+          if ((await element.getComputedRole()) === role) {
+            return element;
+          } else if (
+            (await element.parentElement().getComputedRole()) === role
+          ) {
+            // `aria/Name of button` unfortunately selects the span within,
```

Comment:
What do you mean of `aria/Name of button`? Is it a role button with `aria/{name}`?

It selects it if the button has a span within it, no?

Maybe it's better to give an example of the html structure for this case?